ngi_pipeline/engines/piper_ngi/results_parsers.py

```diff
@@ -33,3 +41,116 @@ def parse_qualimap_coverage(genome_results_file):
+    :raises ValueError: If arguments are incorrect
+    """
+    try:
+        if seqrun_id and fcid and (fcid != seqrun_id.split("_")[3]):
+            raise ValueError(('seqrun_id and fcid both passed as arguments but do not '
+                              'match (seqrun_id: "{}", fcid: "{}")'.format(seqrun_id, fcid)))
+        if seqrun_id:
+            piper_run_id = seqrun_id.split("_")[3]
+        elif fcid:
+            piper_run_id = fcid
+        else:
+            piper_run_id = None
+    except IndexError:
+        raise ValueError('Can\'t parse FCID from run id ("{}")'.format(seqrun_id))
+    # Find all the appropriate files
+    try:
+        os.path.isdir(piper_qc_dir) and os.listdir(piper_qc_dir)
+    except OSError as e:
+        raise OSError('Piper result directory "{}" inaccessible when updating stats to Charon: {}.'.format(piper_qc_dir, e))
+    piper_qc_dir_base = "{}.{}.{}".format(sample_id, (piper_run_id or "*"), sample_id)
+    piper_qc_path = "{}*/".format(os.path.join(piper_qc_dir, piper_qc_dir_base))
+    piper_qc_dirs = glob.glob(piper_qc_path)
+    if not piper_qc_dirs: # Something went wrong, is the sample name with a hyphen or with an underscore ?
+        piper_qc_dir_base = "{}.{}.{}".format(sample_id.replace('_', '-', 1), (piper_run_id or "*"), sample_id.replace('_', '-', 1))
+        piper_qc_path = "{}*/".format(os.path.join(piper_qc_dir, piper_qc_dir_base))
+        piper_qc_dirs = glob.glob(piper_qc_path)
+        if not piper_qc_dirs: # Something went wrong in the alignment or we can't parse the file format
+            raise OSError('Piper qc directories under "{}" are missing or in an unexpected format when updating stats to Charon.'.format(piper_qc_path))
+    mean_autosomal_coverage = 0
+    # Examine each lane and update the dict with its alignment metrics
+    for qc_lane in piper_qc_dirs:
+        genome_result = os.path.join(qc_lane, "genome_results.txt")
+        # This means that if any of the lanes are missing results, the sequencing run is marked as a failure.
+        if not os.path.isfile(genome_result):
+            raise OSError('File "genome_results.txt" is missing from Piper result directory "{}"'.format(piper_qc_dir))
+        # Get the alignment results for this lane
+        mean_autosomal_coverage += parse_qualimap_coverage(genome_result)
+    return mean_autosomal_coverage
+
+
+
+def parse_genotype_concordance(genotype_concordance_file):
+    concordance_data = []
+    gt_values_list = []
+    with open(genotype_concordance_file, 'r') as f:
+        for line in iter(f.readline, ''):
+            header_location = None
+            if line.startswith("#:GATKTable:GenotypeConcordance_Summary"):
+                header_values = [h.strip() for h in f.readline().strip().split('  ') if h.strip()]
+                header_values = [h.lower().replace("-", "_").replace(" ", "_") for h in header_values]
+                GTValue = namedtuple('GTValue', header_values)
```

Comment:
Wasn't @b97pla doing something like this the other day? Something with a config file. `namedtupled` is the way. Maybe I already said that.